atomicfu-gradle-plugin/src/main/kotlin/kotlinx/atomicfu/plugin/gradle/AtomicFUGradlePlugin.kt

```diff
@@ -42,15 +42,38 @@ private const val MIN_SUPPORTED_KGP_VERSION = "1.7.0"
 open class AtomicFUGradlePlugin : Plugin<Project> {
     override fun apply(project: Project) = project.run {
         checkCompatibility()
-        val pluginVersion = rootProject.buildscript.configurations.findByName("classpath")
-            ?.allDependencies?.find { it.name == "atomicfu-gradle-plugin" }?.version
+        // Get the version of atomicfu-gradle-plugin applied to the project:
+        // first try to find it in the root buildscript, then in the buildscript of the current project.
+        // The error is thrown in case atomicfu-gradle-plugin is not applied.
+        val pluginVersion = rootProject.getAGPVersion() ?: this.getAGPVersion()
+        requireNotNull(pluginVersion) { "We were unable to find the version of `kotlinx-atomicfu` plugin applied to the project.\n" +
+                "Please ensure that `kotlinx-atomicfu` plugin is correctly applied.\n" +
+                "Below is an example of Kotlin DSL code that applies `kotlinx-atomicfu` plugin:\n" +
+                "```\n" +
+                "buildscript {\n" +
+                "    repositories {\n" +
+                "        mavenCentral()\n" +
+                "    }\n" +
+                "\n" +
+                "    dependencies {\n" +
+                "      classpath(\"org.jetbrains.kotlinx:atomicfu-gradle-plugin:0.23.2\")\n" +
```

Comment:
> Should we take a version from project settings when building the plugin or something like that?

If you are gonna to decide to do this, then all this hacks to fetch an atomicfu version from a build script will be not needed at all, as you will already know a version of the plugin :)

And IMO, this could be a better approach, that introducing this good error message, because in that case, there will be no need for it as there will be no need to add `atomicfu` to build classpath, and so just applying plugin will be enough :)
It will also resolve issues like #210, #137 or any other issues when users **just** apply plugin in some way (f.e via version catalogs) and not via `buildscript.classpath` (which is, AFAIK, is Gradle legacy and is not suggested approach)

There are multiple ways to do it, from [manual ones](https://discuss.gradle.org/t/how-can-a-custom-gradle-plugin-determine-its-own-version/36761/2) to some existing Gradle plugins like [this](https://github.com/gmazzo/gradle-buildconfig-plugin) (which works pretty fine at my side)

Reply:
sorry, the comment should be left here: https://github.com/Kotlin/kotlinx-atomicfu/pull/405#discussion_r1506167680